packages/circuits/src/shuffle.circom

```diff
@@ -0,0 +1,35 @@
+pragma circom 2.0.0;
+
+include "./merkle.circom";
+
+template Shuffle(levels) {
+
+    signal input initialDeck[2**levels];
+    signal input finalDeck[2**levels];
+    signal input deckPredicate[2**levels];
```

Comment:
Would be good to define (comment) `deckPredicate` here.

I'm also not in love with the name (also in the other proof). Maybe `initialDeckWeights`? Open to suggestions!

Reply:
I think `deckWeights` might make sense, since it's not really "initial" per se as the weights are dependent on the drawn indices.